lib/elixir/lib/kernel.ex

```diff
@@ -261,6 +261,8 @@ defmodule Kernel do
   @doc """
   Returns the head of a list. Raises `ArgumentError` if the list is empty.
 
+  It works with improper lists.
```

Comment:
I think our policy is that it doesn't hurt repeating what is in the spec in the documentation. Especially because reading the specs takes time and is more for advanced developers.